### PR TITLE
oci: fixes to properly handle container stop action

### PR DESCRIPTION
**Pull request by runcom**
@mrunalp PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

File: oci/oci.go [hunk 1]
```diff
@@ -588,7 +575,10 @@ func (r *Runtime) StopContainer(c *Container, timeout int64) error {
 	select {
 	case <-done:
 		return nil
-	case <-time.After(time.Duration(timeout) * time.Second):
+	case <-ctx.Done():
+		close(chControl)
+		return ctx.Err()
+	case <-time.After(timeout):
```
Comment (TomSweeneyRedHat) [inline oci/oci.go]: The "time.Duration(timeout) * time.Second" is equivalent to timeout?

Comment (TomSweeneyRedHat) [reply oci/oci.go]: Ah, I see, you changed it form an int.  NVM!

File: server/sandbox_remove.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func (s *Server) RemovePodSandbox(ctx context.Context, req *pb.RemovePodSandboxR
 		if !sb.Stopped() {
 			cState := s.Runtime().ContainerStatus(c)
 			if cState.Status == oci.ContainerStateCreated || cState.Status == oci.ContainerStateRunning {
-				if err := s.Runtime().StopContainer(c, -1); err != nil {
+				if err := s.Runtime().StopContainer(ctx, c, 10); err != nil {
```
Comment (TomSweeneyRedHat) [inline server/sandbox_remove.go]: Should the 10 be in a conf file somewhere on the very off chance we ever want to tweak it live?

File: server/sandbox_stop.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ func (s *Server) StopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 	for _, c := range containers {
 		cStatus := s.Runtime().ContainerStatus(c)
 		if cStatus.Status != oci.ContainerStateStopped {
-			if err := s.Runtime().StopContainer(c, -1); err != nil {
+			if err := s.Runtime().StopContainer(ctx, c, 10); err != nil {
```
Comment (TomSweeneyRedHat) [inline server/sandbox_stop.go]: Ditto 10 to a conf file comment.

File: server/sandbox_remove.go (hunk 2 above)
Comment (runcom) [reply server/sandbox_remove.go]: nope, nobody should tweak this and this PR is the same as #1010 so no need to re-review I guss

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**